streams/src/main/java/org/apache/kafka/streams/processor/internals/StoreChangelogReader.java

```diff
@@ -444,6 +451,8 @@ public void restore(final Map<TaskId, Task> tasks) {
                 final Set<TaskId> corruptedTasks = new HashSet<>();
                 e.partitions().forEach(partition -> corruptedTasks.add(changelogs.get(partition).stateManager.taskId()));
                 throw new TaskCorruptedException(corruptedTasks, e);
+            } catch (final InterruptException interruptException) {
+                throw interruptException;
```

Comment:
QQ: will `InterruptException` be thrown even without this?

Reply:
Yes, it will. However, `InterruptException` is a `KafkaException`, thus it would be caught the `catch`-clause below and rethrown as a `StreamsException`. I thought it would be easier to directly catch an `InterruptException` in `DefaultStateUpdater` instead of catching a `StreamsException`, unwrap it, verify if it is an `InterruptException`, and if not rethrow it.